src/cpp/rtps/builtin/discovery/participant/PDP.cpp

```diff
@@ -481,9 +483,9 @@ bool PDP::has_reader_proxy_data(const GUID_t& reader)
     {
         if (pit->m_guid.guidPrefix == reader.guidPrefix)
         {
-            for (ReaderProxyData* rit : pit->m_readers)
+            for (auto rp : *pit->m_readers)
             {
-                if (rit->guid() == reader)
+                if (rp.second->guid() == reader)
```

Comment:
We could use `pit->m_readers->find` here

Reply:
In fact, we can just do `return pit->m_readers->find(reader.entityId) != pit->m_readers->end();`

Reply:
agreed, we can speed up by using hash table capabilities:
```
 ProxyHashTable<ReaderProxyData> & readers = *pit->m_readers;
 return readers.find(reader.entityId) != readers.end();
```